### ISSUE TITLE
cycling through `kubeapps up` and `down` leaves roles in bad state

### ISSUE THREAD

**Issue by sebgoa** [kubeapps/kubeapps#69]
I did a `kubeapps up` , then a `kubeapps down` everything seemed to have been deleted.
then did a `kubeapps up` again, everything re-started, but the kubeless UI got in a bad state and could not read the pods corresponding to a function.

The logs of the kubeless ui showed:

```
Wed, 29 Nov 2017 23:54:01 GMT app:server Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services \"foo\" is forbidden: User \"system:serviceaccount:kubeapps:kubeless-ui\" cannot get services in the namespace \"default\": Unknown user \"system:serviceaccount:kubeapps:kubeless-ui\"","reason":"Forbidden","details":{"name":"foo","kind":"services"},"code":403}

Wed, 29 Nov 2017 23:54:01 GMT app:server Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services \"foo\" is forbidden: User \"system:serviceaccount:kubeapps:kubeless-ui\" cannot get services in the namespace \"default\": Unknown user \"system:serviceaccount:kubeapps:kubeless-ui\"","reason":"Forbidden","details":{"name":"foo","kind":"services"},"code":403}
```

**Comment by ngtuna** [comment]
@sebgoa Have you tried that action on kubeless-ui before ? I don't think cycling `kubeapps up/down` is the main cause. Looking into the ClusterRoleBinding for the ServiceAccount kubeless-ui I see:

```
$ kubectl get clusterrolebinding kubeless-ui-editor -o yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  annotations:
    kubecfg.ksonnet.io/garbage-collect-tag: bitnami/kubeapps
  creationTimestamp: 2017-11-30T08:18:37Z
  labels:
    name: kubeless-ui-editor
  name: kubeless-ui-editor
  resourceVersion: "136536"
  selfLink: /apis/rbac.authorization.k8s.io/v1beta1/clusterrolebindings/kubeless-ui-editor
  uid: 10bbc643-d5a7-11e7-97d1-42010a940086
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kubeless-editor
subjects:
- kind: ServiceAccount
  name: kubeless-ui
  namespace: kubeapps

$ kubectl get clusterrole kubeless-editor -o yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  annotations:
    kubecfg.ksonnet.io/garbage-collect-tag: bitnami/kubeapps
  creationTimestamp: 2017-11-29T04:50:58Z
  deletionGracePeriodSeconds: 0
  deletionTimestamp: 2017-11-30T08:18:06Z
  finalizers:
  - foregroundDeletion
  labels:
    name: kubeless-editor
  name: kubeless-editor
  resourceVersion: "136262"
  selfLink: /apis/rbac.authorization.k8s.io/v1beta1/clusterroles/kubeless-editor
  uid: e4201eb6-d4c0-11e7-97d1-42010a940086
rules:
- apiGroups:
  - k8s.io
  resources:
  - functions
  verbs:
  - get
  - list
  - watch
  - create
  - patch
```

The kubeless-ui can only do actions on functions object I guess.

**Comment by sebgoa** [comment]
ouh interesting...

in the kubeless-ui you can get the logs of the pod. so the UI actually looks for the pod that makes the function and then streams the logs back...

it can also call the function (access the service).

This happened when I created GKE clusters with only RBAC enabled (no legacy auth).

So we need to tweak the RBAC for the kubeless-ui it seems

**Comment by sebgoa** [comment]
@ngtuna actually check the kubeless UI manifest:

https://github.com/kubeless/kubeless-ui/blob/master/k8s.yaml

it looks like the rBAC has more than the functions, seems the kubeapps/manifest did not take that into account.

**Comment by sebgoa** [comment]
fixing this through: https://github.com/kubeapps/manifest/pull/30

**Comment by anguslees** [comment]
> it looks like the rBAC has more than the functions, seems the kubeapps/manifest did not take that into account.

Yep, `kubeless-ui.jsonnet` in manifests hasn't been updated since kubeless/kubeless-ui#35 was merged.  Looking at the diff now(*) these RBAC additions have been the only relevant changes upstream.

(*) https://github.com/kubeless/kubeless-ui/compare/96231f8640f39cb6b7e658c5d50a58871bccb0a6...master#diff-8c7803327cbf2efb920f4692d702d48a

**Comment by ngtuna** [comment]
closed via #72
